File: 3347-distribute-elements-into-two-arrays-i/distribute-elements-into-two-arrays-i.py
class Solution:
    def resultArray(self, nums: List[int]) -> List[int]:
        n = len(nums)
        l , r = nums[0] , nums[1]
        nums[1] *= -1

        for i in range(2 , n):
            if l > r :
                l = nums[i]
            else:
                r = nums[i]
                nums[i] *= -1
        
        res = []
        for i in range(n):
            if nums[i] > 0:
                res.append(nums[i])
        for i in range(n):
            if nums[i] < 0 :
                res.append(-nums[i])
        
        return res

File: 3347-distribute-elements-into-two-arrays-i/distribute-elements-into-two-arrays-i.cpp
class Solution {
public:
    vector<int> resultArray(vector<int>& nums) {
        int l = nums[0] , r = nums[1] , n = nums.size() , i;
        nums[1] *= -1;
        for (i = 2 ; i < n ; i++){
            if (l > r){
                l = nums[i];
            }
            else{
                r = nums[i];
                nums[i] *= -1;
            }
        }
        std::vector<int> res;
        for (i = 0 ; i < n ; i++){
            if (nums[i] > 0){
                res.push_back(nums[i]);
            }
        }
        for (i = 0 ; i < n ; i++){
            if (nums[i] < 0){
                res.push_back(-nums[i]);
            }
        }
        return res;
    }
};